scripts/convert_data.py

```diff
@@ -0,0 +1,243 @@
+    :return: list of named tuples for each line of the data representing a single named entity token
+    """
+    if len(bsf_data.strip()) == 0:
+        return []
+
+    ln_ptrn = re.compile(r'(T\d+)\s(\w+)\s(\d+)\s(\d+)\s(.+?)(?=T\d+\s\w+\s\d+\s\d+|$)', flags=re.DOTALL)
+    result = []
+    for m in ln_ptrn.finditer(bsf_data.strip()):
+        bsf = BsfInfo(m.group(1), m.group(2), int(m.group(3)), int(m.group(4)), m.group(5).strip())
+        result.append(bsf)
+    return result
+
+
+def convert_bsf_in_folder(src_dir_path: str, dst_dir_path: str, converter: str = 'beios',
+                          doc_delim: str = '\n', train_test_split_file: str = None) -> None:
+    """
+
+    :param doc_delim: delimiter to be used between documents
+    :param src_dir_path: path to directory with BSF marked files
+    :param dst_dir_path: where to save output data
+    :param converter: `beios` or `iob` output formats
+    :param train_test_split_file: path to file containing train/test lists of file names
+    :return:
+    """
+    # following 2 constants need to comply with stanza naming for corpus and language
+    corpus_name = 'Ukrainian-languk'
+
+    ann_path = os.path.join(src_dir_path, '*.tok.ann')
+    ann_files = glob.glob(ann_path)
+    ann_files.sort()
+
+    tok_path = os.path.join(src_dir_path, '*.tok.txt')
+    tok_files = glob.glob(tok_path)
+    tok_files.sort()
+
+    corpus_folder = os.path.join(dst_dir_path, corpus_name)
+    if not os.path.exists(corpus_folder):
+        os.makedirs(corpus_folder)
+
+    if len(ann_files) == 0 or len(tok_files) == 0:
+        log.warning(f'Token and annotation files are not found at specified path {ann_path}')
+        return
+    if len(ann_files) != len(tok_files):
+        log.warning(
+            f'Mismatch between Annotation and Token files. Ann files: {len(ann_files)}, token files: {len(tok_files)}')
+
+    train_set = []
+    dev_set = []
+    test_set = []
+
+    data_sets = [train_set, dev_set, test_set]
+    split_weights = (8, 1, 1)
+
+    if train_test_split_file is not None:
+        train_names, dev_names, test_names = read_languk_train_test_split(train_test_split_file, 0)
+
+    log.info(f'Found {len(tok_files)} files in data folder "{src_dir_path}"')
+    for (tok_fname, ann_fname) in tqdm(zip(tok_files, ann_files), total=len(tok_files), unit='file'):
+        if tok_fname[:-3] != ann_fname[:-3]:
+            tqdm.write(f'Token and Annotation file names do not match ann={ann_fname}, tok={tok_fname}')
+            continue
+
+        with open(tok_fname) as tok_file, open(ann_fname) as ann_file:
+            token_data = tok_file.read()
+            ann_data = ann_file.read()
+            out_data = convert_bsf(token_data, ann_data, converter)
+
+            if train_test_split_file is None:
+                target_dataset = choices(data_sets, split_weights)[0]
+            else:
+                target_dataset = train_set
+                fkey = os.path.basename(tok_fname)[:-4]
```

Comment:
`os.path.splitext` seems to be a more robust alternative

Reply:
Makes sense. Fixed